projects.html

```diff
@@ -19,21 +19,8 @@ <h1 id="{{section.name | downcase | replace:' ','_' }}">{{section.name}}</h1>
                         {% endif %}
                     </span>
                     <span class="project__social">
-                    {% if project.video %}
-                        <a href="{{project.video}}" target="_blank"><i class="fa fa-youtube-play"></i></a>
-                    {% endif %}
-                    {% if project.slides %}
-                        <a href="{{project.slides}}" target="_blank"><i class="fa fa-area-chart"></i></a>
-                    {% endif %}
-                    {% if project.paper %}
-                        <a href="{{project.paper}}" target="_blank"><i class="fa fa-book"></i></a>
-                    {% endif %}
-                    {% if project.github %}
-                        <a href="https://github.com/{{project.github}}" target="_blank"><i class="fa fa-github"></i></a>
-                    {% endif %}
-                    {% if project.twitter %}
-                        <a href="https://twitter.com/{{project.twitter}}" target="_blank"><i class="fa fa-twitter"></i></a>
-                    {% endif %}
+                    {% assign subsection = project %}
```

Comment:
This assignment translates the domain-specific 'projects' into the generic 'subsections' so that the following `include` statement will function properly.

Inlining the assignment in the following `include` would be cleaner/more robust, and it [appears possible](https://jekyllrb.com/docs/includes/#passing-parameters-to-includes) to set a default value when no explicit assignment is made. (E.g. for cases without a domain-specific name for their subsections like `undergrad.yml` ). 

After an hour of tinkering with it, though, I haven't made it work. I'm calling it for now. If this feels important to this PR, I'll try again next week. Otherwise, I can write this up as a possible `good first issue`.

LMK, Chris

Reply:
@ChrisKeefe, this is truly _fantastic_! :tada: :balloon:! _Excellent work_!


> Inlining the assignment in the following `include` would be cleaner/more robust, and it [appears possible](https://jekyllrb.com/docs/includes/#passing-parameters-to-includes) to set a default value when no explicit assignment is made.

I agree - this would be a little more "declarative." I poked around a bit and this _appears_ to be working:

```diff
diff --git a/_includes/functions/get_social_icons.html b/_includes/functions/get_social_icons.html
index f56c22a..08b490c 100644
--- a/_includes/functions/get_social_icons.html
+++ b/_includes/functions/get_social_icons.html
@@ -1,5 +1,5 @@
 {% for valid_tag in site.data.social %}
-  {% if subsection[valid_tag.tag] %}
+  {% if include.social[valid_tag.tag] %}
   <a href="{{valid_tag.base_url}}{{subsection[valid_tag.tag]}}" target="_blank"><i class="{{valid_tag.css_class}}"></i></a>
   {% endif %}
 {% endfor %}
\ No newline at end of file
diff --git a/undergrad.html b/undergrad.html
index 931c2fe..ec68081 100644
--- a/undergrad.html
+++ b/undergrad.html
@@ -19,7 +19,7 @@ color: MediumVioletRed
                         {% endif %}
                     </span>
                     <span class="ug__social">
-                    {% include functions/get_social_icons.html %}
+                    {% include functions/get_social_icons.html social=subsection %}
                     </span>
                 </div>
                 <div class="ug__summary">
```

Reply:
Thanks for the assist - I thought I'd tried everything, but clearly I was wrong! :joy:
One little tweak from this diff, and she's firing on all three cylinders.